projects/plugins/jetpack/extensions/blocks/tock/edit.js

```diff
@@ -0,0 +1,154 @@
+import { useBlockProps, BlockControls } from '@wordpress/block-editor';
+import {
+	SandBox,
+	TextControl,
+	ToolbarGroup,
+	ToolbarButton,
+	Notice,
+	Popover,
+	Button,
+} from '@wordpress/components';
+import { createInterpolateElement, useState } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { keyboardReturn } from '@wordpress/icons';
+import { ENTER, ESCAPE } from '@wordpress/keycodes';
+
+import './editor.scss';
+
+const ToolbarControls = ( { isOpen, open, close } ) => (
+	<ToolbarGroup>
+		<ToolbarButton
+			className="components-toolbar__control"
+			label={ __( 'Edit URL', 'jetpack' ) }
+			icon="edit"
+			onClick={ () => ( isOpen ? close() : open() ) }
+		/>
+	</ToolbarGroup>
+);
+
+const UrlPopover = ( { tockUrl, setEditedUrl, popoverAnchor, setUrl, cancel } ) => {
+	const handleSubmitOrCancel = event => {
+		const { keyCode } = event;
+
+		if (
+			keyCode === ENTER &&
+			'' !== tockUrl // Disallow submitting empty values.
+		) {
+			event.preventDefault();
+			setUrl();
+		}
+
+		if ( keyCode === ESCAPE ) {
+			event.preventDefault();
+			cancel();
+		}
+	};
+
+	return (
+		<Popover anchor={ popoverAnchor }>
+			<div className="jetpack-tock-url-input-wrapper">
+				<TextControl
+					placeholder="Add Tock business name"
```

Comment:
Could you make this translatable?

```suggestion
					placeholder={ __( 'Add Tock business name', 'jetpack' ) }
```